esmvalcore/preprocessor/_multimodel.py

```diff
@@ -201,158 +185,88 @@ def _put_in_cube(template_cube, cube_data, statistic, t_axis):
     return stats_cube
 
 
-def _datetime_to_int_days(cube):
-    """Return list of int(days) converted from cube datetime cells."""
-    cube = _align_yearly_axes(cube)
-    time_cells = [cell.point for cell in cube.coord('time').cells()]
-
-    # extract date info
-    real_dates = []
-    for date_obj in time_cells:
-        # real_date resets the actual data point day
-        # to the 1st of the month so that there are no
-        # wrong overlap indices
-        real_date = datetime(date_obj.year, date_obj.month, 1, 0, 0, 0)
-        real_dates.append(real_date)
-
-    # get the number of days starting from the reference unit
-    time_unit = cube.coord('time').units.name
-    time_offset = _get_time_offset(time_unit)
-    days = [(date_obj - time_offset).days for date_obj in real_dates]
-
-    return days
-
-
-def _align_yearly_axes(cube):
-    """Perform a time-regridding operation to align time axes for yr data."""
-    years = [cell.point.year for cell in cube.coord('time').cells()]
-    # be extra sure that the first point is not in the previous year
-    if 0 not in np.diff(years):
-        return regrid_time(cube, 'yr')
-    return cube
-
-
-def _get_overlap(cubes):
+def _unify_time_coordinates(cubes):
     """
-    Get discrete time overlaps.
-
-    This method gets the bounds of coord time
-    from the cube and assembles a continuous time
-    axis with smallest unit 1; then it finds the
-    overlaps by doing a 1-dim intersect;
-    takes the floor of first date and
-    ceil of last date.
-    """
-    all_times = []
-    for cube in cubes:
-        span = _datetime_to_int_days(cube)
-        start, stop = span[0], span[-1]
-        all_times.append([start, stop])
-    bounds = [range(b[0], b[-1] + 1) for b in all_times]
-    time_pts = reduce(np.intersect1d, bounds)
-    if len(time_pts) > 1:
-        time_bounds_list = [time_pts[0], time_pts[-1]]
-        return time_bounds_list
+    Make sure all cubes' use the same standard calendar and time units.
 
+    Cubes may have different calendars. This function extracts the date
+    information from the cube and re-constructs a default calendar,
+    resetting the actual dates to the 15th of the month or 1st of july for
+    yearly data (consistent with `regrid_time`), so that there are no
+    mismatches in the time arrays.
 
-def _slice_cube(cube, t_1, t_2):
+    Doesn't work for (sub)daily data, because different calendars may have
```

Comment:
This could work just fine if all calendars are the same already, in that case there would be no need to change the calendar or time points

Reply:
Fair point. I'll print a warning instead and attempt to continue.

Reply:
It would be better not to change the calendars if no change is needed.

Reply:
Okay, I now made it such that it will first check if the calendars are consistent. If they are consistent, they will not be changed. Otherwise, they will be changed to a default calendar.

Reply:
Much better, because now the function can hopefully be used to compute e.g. ensemble statistics without unexpected results. Maybe we should actually move the whole changing the time coordinate to the `regrid_time` preprocessor, that would be more consistent with what we do for the horizontal and vertical coordinates and would make the process more visible to users of the recipe and the implementation of the multi model statistics function much easier. I've created this issue  #744.